pkg/appmanager/resourceConfig_test.go

```diff
@@ -1371,240 +1371,387 @@ var _ = Describe("Resource Config Tests", func() {
+			Expect(resourceConfig.Policies[0].Rules[2].Actions).To(Equal([]*action{rule1.Actions[0], rule2.Actions[0]}))
+			Expect(resourceConfig.Policies[0].Rules[2].Conditions).To(Equal(rule1.Conditions))
+
+			// Verify rule6 (longest URI) was sorted behind url-rewrite but ahead of shorter URIs
+			Expect(resourceConfig.Policies[0].Rules[3].Name).To(Equal(rule6.Name))
+			Expect(resourceConfig.Policies[0].Rules[3].Ordinal).To(Equal(3))
+			Expect(resourceConfig.Policies[0].Rules[3].Actions).To(Equal(rule6.Actions))
+			Expect(resourceConfig.Policies[0].Rules[3].Conditions).To(Equal(rule6.Conditions))
+
+			// Verify last rule is the shorted URI rule
+			Expect(resourceConfig.Policies[0].Rules[4].Name).To(Equal(rule4.Name))
+			Expect(resourceConfig.Policies[0].Rules[4].Ordinal).To(Equal(4))
+			Expect(resourceConfig.Policies[0].Rules[4].Actions).To(Equal(rule4.Actions))
+			Expect(resourceConfig.Policies[0].Rules[4].Conditions).To(Equal(rule4.Conditions))
+
+			// Verify calling UnmergeRule on an unmerged rule does nothing
+			result := resourceConfig.UnmergeRule(rule7.Name, mergedRulesMap)
 			Expect(len(resourceConfig.Policies[0].Rules)).To(Equal(5))
-			Expect(resourceConfig.Policies[0].Rules[4].Name).To(Equal(rule8.Name))
-			Expect(resourceConfig.Policies[0].Rules[4].Conditions).To(Equal(rule8.Conditions))
-			Expect(resourceConfig.Policies[0].Rules[4].Actions).To(Equal([]*action{rule8.Actions[0]}))
-			Expect(mergedRulesMap["test-virtual"][rule4.Name]).To(Equal(mergedRuleEntry{}))
-			Expect(mergedRulesMap["test-virtual"][rule4.Name]).To(Equal(mergedRuleEntry{}))
+			Expect(result).To(BeFalse())
 
-			// Unmerge rule2
-			resourceConfig.UnmergeRule(rule2.Name, mergedRulesMap)
+			// Verify calling UnmergeRule on an already unmerged rule does nothing
+			result = resourceConfig.UnmergeRule(rule3.Name, mergedRulesMap)
 			Expect(len(resourceConfig.Policies[0].Rules)).To(Equal(5))
-			Expect(resourceConfig.Policies[0].Rules[3].Name).To(Equal(rule1.Name))
-			Expect(resourceConfig.Policies[0].Rules[3].Conditions).To(Equal(rule1.Conditions))
-			Expect(resourceConfig.Policies[0].Rules[3].Actions).To(Equal([]*action{rule1.Actions[0]}))
-			Expect(mergedRulesMap["test-virtual"][rule1.Name]).To(Equal(mergedRuleEntry{}))
-			Expect(mergedRulesMap["test-virtual"][rule2.Name]).To(Equal(mergedRuleEntry{}))
-			Expect(len(mergedRulesMap["test-virtual"])).To(Equal(0))
-
-			// Unmerge rule3
-			result := resourceConfig.UnmergeRule(rule2.Name, mergedRulesMap)
 			Expect(result).To(BeFalse())
 
-			// Unmerge rule7 again
-			result = resourceConfig.UnmergeRule(rule7.Name, mergedRulesMap)
-			Expect(result).To(BeFalse())
+			// Unmerge mergee rule2
+			resourceConfig.UnmergeRule(rule2.Name, mergedRulesMap)
+			Expect(len(mergedRulesMap["test-virtual"])).To(Equal(0))
+			Expect(len(resourceConfig.Policies[0].Rules)).To(Equal(5))
 
-			// Unmerge rule1
+			// Verify first rule has not changed
+			Expect(len(resourceConfig.Policies[0].Rules)).To(Equal(5))
```

Comment:
This case was just checked, no need to do it again.

Reply:
Removing.